tests/components/nextcloud/test_sensor.py

```diff
@@ -0,0 +1,89 @@
+"""The tests for the nextcloud sensor component."""
+
+from unittest import TestCase
+
+from homeassistant.components.nextcloud.sensor import NextcloudSensor
+
+from tests.common import get_test_home_assistant, setup_component
+
+API_RESPONSE = {"nextcloud_system_version": "17.0.3.1"}
+
+VALID_CONFIG = {
+    "sensor": {
+        "platform": "nextcloud",
+        "url": "https://nextcloud.sample.org",
+        "username": "username",
+        "password": "password",
+    }
+}
+
+INVALID_CONFIG = {
+    "sensor": {
+        "platform": "nextcloud",
+        "url": [],  # Url as empty list to invalidate configuration
+        "username": "username",
+        "password": "password",
+    }
+}
+
+
+class testSensorSetup(TestCase):
+    """Test the setup of the Nextcloud sensors."""
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.hass = get_test_home_assistant()
+        self.entities = []
+
+    def tearDown(self):
+        """Tear down after tests."""
+        self.hass.stop()
+
+    def test_valid_config(self):
+        """Test setup using a valid configuration."""
+        setup_component(self.hass, "sensor", VALID_CONFIG)
+
+    def test_invalid_config(self):
+        """Test setup using an invalid configuration."""
+        setup_component(self.hass, "sensor", INVALID_CONFIG)
+
+
+class test_NextcloudSensor(TestCase):
+    """Tests the Nextcloud sensor."""
+
+    def setUp(self):
+        """Set up things to be run when tests are started."""
+        self.ncs = NextcloudSensor("nextcloud_system_version")
```

Comment:
We don't want to instantiate the entity directly but let the our platform do that after setting up the component via the core. Use `async_setup_component` to set up the component and platform and let the platform add an entity. Then assert the state of the entity via the state machine.

Reply:
I think this is way over my head currently :/
I'm self-taught and not up to speed on pytest or async yet.
I'll see what i can do.

Reply:
Tests are not a requirement for merging this PR. It's ok to remove the tests and exclude the modules in `.coveragerc` for now. Only if adding a config flow are tests of the config flow required.

Reply:
Ok, removed for now.